### ISSUE TITLE
liveness or readiness probes ignore timeout

### ISSUE THREAD

**Issue by MrYHM** [kubernetes/kubernetes#128408]
### What happened?

When executing livenessProbe exec command, timeoutSeconds does not work.
timeoutSeconds: 5
periodSeconds: 30
exec command: /opt/entrypoint.sh healthcheck ,healthcheck will request a url without timeout. When this URL does not respond, a liveness request is received approximately every 2 minutes.

### What did you expect to happen?

When this URL does not respond, liveness request timed out after 5 seconds.

### How can we reproduce it (as minimally and precisely as possible)?

Maybe sleep 120s in command, timeoutSeconds: 5, periodSeconds: 30

### Anything else we need to know?

livenessProbe config:
`livenessProbe:
        exec:
          command:
            - /opt/entrypoint.sh
            - healthcheck
        initialDelaySeconds: 30
        timeoutSeconds: 5
        periodSeconds: 30
        successThreshold: 1
        failureThreshold: 12
      readinessProbe:
        exec:
          command:
            - /opt/entrypoint.sh
            - healthcheck
        initialDelaySeconds: 30
        timeoutSeconds: 5
        periodSeconds: 30
        successThreshold: 1
        failureThreshold: 12`

### Kubernetes version

<details>

```
Server Version: version.Info{Major:"1", Minor:"29+", GitVersion:"v1.29.8-eks-a737599", GitCommit:"3277d87d88d0bf66b6368ce57e49b2f2aab01b0d", GitTreeState:"clean", BuildDate:"2024-08-26T21:27:41Z", GoVersion:"go1.22.5", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
AWS eks
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by xyz-li** [comment]
If you haven't change `runtimeRequestTimeout`, then the request time out is about 125 seconds 

https://github.com/kubernetes/kubernetes/blob/b8e20b74ddc9d0f072436e569877429ea6efa799/pkg/kubelet/apis/config/types.go#L259-L261

https://github.com/kubernetes/kubernetes/blob/b8e20b74ddc9d0f072436e569877429ea6efa799/staging/src/k8s.io/cri-client/pkg/remote_runtime.go#L486-L496

**Comment by MrYHM** [comment]
@xyz-li Yes, I see the same phenomenon as you described(time out is 120 seconds + timeoutSeconds), what should I do? Thank you.

**Comment by xyz-li** [comment]
Can you use  `HTTP` probe? Or you can reduce  `runtimeRequestTimeout`
@MrYHM

**Comment by MrYHM** [comment]
@xyz-li 
This application does not provide external services and does not have an HTTP server.
I added a 5 second timeout to the entrypoint.sh script, and that's it for now.
Thank you.

**Comment by saschagrunert** [comment]
Do we need to enhance the documentation about this? https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes

**Comment by MrYHM** [comment]
@saschagrunert 
If the exec cammond timeout is runtimeRequestTimeout + timeoutSeconds, then please update the document. Thanks.

**Comment by hshiina** [comment]
Can you see if `ExecProbeTimeout` feature gate is enabled in your environment just in case?
https://github.com/kubernetes/kubernetes/blob/cf480a3a1a9cb22f3439c0a7922822d9f67f31b5/pkg/probe/exec/exec.go#L74-L80

**Comment by AnishShah** [comment]
sig-node CI Meeting note:

/triage needs-information

**Comment by MrYHM** [comment]
@hshiina I am using EKS, and ExecProbeTimeout should be enabled by default.

**Comment by hshiina** [comment]
If I understand correctly, `120 seconds + timeoutSeconds` is communication timeout used by the runtime client. The runtime should handle `timeoutSeconds`. This is [containerd code](https://github.com/containerd/containerd/blob/bcc810d6b9066471b0b6fa75f557a15a1cbf31bb/pkg/cri/sbserver/container_execsync.go#L205-L226
) though I don't know what runtime is used in EKS.

**Comment by hshiina** [comment]
@MrYHM 
Did you see an event whose message contained `probe errored`?  This event is often observed when a probe doesn't work as expected:
https://github.com/kubernetes/kubernetes/blob/86e25a07e279516eb13af436ca5706b1806ea604/pkg/kubelet/prober/prober.go#L105

**Comment by xuegege5290** [comment]
https://github.com/kubernetes/kubernetes/blob/b8e20b74ddc9d0f072436e569877429ea6efa799/pkg/kubelet/apis/config/types.go#L259-L261
If the runtimeRequestTimeout is included, should the annotation be modified? It defines the timeout for all runtime requests except long-running ones such as image pulls, logs, exec, and attach.
